### PR TITLE
Convert GSE164677 to TPM and add scRNA GSE155446

### LINKED ISSUE #74: Additional 10X 3' scRNA data
The single cell data we currently have (https://www.nature.com/articles/s41586-019-1434-6) was generated using Smart-seq2, which is less sparse than 10X 3' scRNA-seq. What if we found an additional data set to test on? If we're going to make claims about single cell data, it would be nice for the claims to be explored using the most prevalent (I assume) type of single cell data.

### PR DESCRIPTION
**Pull request by envest**
Closes #74 and #77

This PR uses a different input file for GSE164677 (TPM), and it adds more scRNA data (GSE155446) and converts that to TPM.

`00-download_data.sh`

- Add provisions for new scRNA GSE155446

`01-gather_metadata.R`

- Use `here::here()`
- Update how to `clean_mb_subgroups()`
- Use getGEO to read in GSE164677 metadata
- Add GSE155446 metadata and combine with other pseudobulk data

`02-gather_data.R`

- Use `here::here()`
- Make separate pseudobulk output directories files
- Add GSE155446 raw count file reading and TPM conversion

`data/data_sources.tsv`

- Update GSE164677 and add GSE155446 files

`data/md5_check_sums.tsv`

- Update check sums

`data/split_scRNA.sh`

- Script to split a scRNA data file by sample when formatted with `sample_barcode` columns

`processed_data/pseudobulk_sce/.gitkeep`

- Removed since `processed_data/pseudobulk_sce` directory now recursively produced by `02-gather_data.R`

**For the reviewer:** I have a specific question in `02-gather_data.R` (detailed comments about this question are in the script). There is some mismatch between the metadata sample names and the file names for the new scRNA data (GSE155446). Only two files are affected, and we could just not include them. The question is: make a decision now to include/exclude them upstream, or save the decision for later and include/exclude them downstream. I prefer to decide now because I'll forget to make a decision later and would like the whole project to be consistent. Thanks!

### CODE REVIEW

File: 02-gather_data.R [hunk 1]
```diff
@@ -225,21 +245,106 @@ for (sample_iter in 1:length(sample_accession_ids)) {
...
+# save GSE119926 pseudobulk df object
+readr::write_tsv(GSE119926_pseudobulk_df,
+                 GSE119926_pseudobulk_genex_df_output_filepath)
+
+# GSE155446, samples have already been split by accession
+# Two sample names are mismatched between metadata and data files
+# Metadata sample 966-recurrence likely matches data file 966-2.tsv
+# Metadata sample 934-repeat MAY match data file 943.tsv, since that's the only remaining non-matching file
+# Including those samples at this stage does not impact downstream analysis.
+# They can be removed later but it would be nice to settle the issue here.
```
Comment (jaclyn-taroni) [inline 02-gather_data.R:284]: I would probably remove them to be conservative.

File: 02-gather_data.R [hunk 2]
```diff
@@ -225,21 +245,106 @@ for (sample_iter in 1:length(sample_accession_ids)) {
...
+
+GSE155446_pseudobulk_metadata <- pseudobulk_metadata |>
+  dplyr::filter(study == "GSE155446") |>
+  #dplyr::filter(sample_accession != "966-recurrence") |> # can be un/commented
+  #dplyr::filter(sample_accession != "943") |> # can be un/commented
+  dplyr::mutate(data_file_prefix = dplyr::case_when(sample_accession == "966-recurrence" ~ "966-2", # can be un/commented
+                                                    sample_accession == "934-repeat" ~ "943", # can be un/commented
+                                                    .default = sample_accession))
```
Comment (jaclyn-taroni) [inline 02-gather_data.R:292]: So I would make the opposite choice about what to uncomment 😄

**Review by jaclyn-taroni — APPROVED** [review]
Looks good. I would drop the two samples with mismatched metadata files to be safe.

**Comment by envest** [comment]
Thanks! I ended up removing the unsure samples from the metadata as well, since we also use that in the downstream analysis.

**Pull request merged**